doc/adr/0016- end-to-end-testing-strategy.rst

```diff
@@ -0,0 +1,274 @@
+.. _end_to_end_testing_strategy:
+
+ADR 16: End-to-end testing strategy for Plutus, cardano-ledger-api and cardano-node-client
+==========================================================================================
+
+Date: 2022-12-02
+
+Author(s)
+---------
+
+james-iohk <james.browning@iohk.io>
+
+Status
+------
+
+Draft
+
+Context
+-------
+
+End-to-end testing of `Plutus Core <https://github.com/input-output-hk/Plutus/>`_ functionality is
+currently performed by a combination of automation and exploratory approaches, both are performed
+on public `preview` and `pre-prod` testnets using a real node.
+Automation test scenarios for Plutus are currently being run as part of the wider `cardano-node-tests
+<https://github.com/input-output-hk/cardano-node-tests/>`_ test suite, which uses a Python wrapper
+for ``cardano-cli``.
+Those tests focus on general ledger/node/cli functionality and only cover a few key scenarios for
+Plutus functionality, such as TxInfo and SECP256k1
+builtins.
+
+There is also ongoing development work to separate the functionality of `cardano-api
+<https://github.com/input-output-hk/cardano-node/tree/master/cardano-api>`_ out into two packages:
+
+* `cardano-ledger-api
+  <https://github.com/input-output-hk/cardano-ledger/tree/master/libs/cardano-ledger-api>`_ handles
+  the building and balancing of transactions.
+
+* ``cardano-node-client`` will live in `cardano-node
+  <https://github.com/input-output-hk/cardano-node>`_ and handle the submitting of balanced
+  transactions and querying the ledger state.
+
+Both of these packaged are in early stage development and will require end-to-end test coverage.
+
+This document outlines the decisions and arguments for an **additional** approach to end-to-end test
+automation using a framework written in Haskell.
+
+The exploratory testing approach is not in the scope of this document.
+
+Decision
+--------
+
+* We will create a new end-to-end testing framework written in Haskell that'll live in a new
```

Comment:
Also, if Plutus Tools team is going to be the main maintainers, than I think it would make more sense to keep it in plutus-apps.

Reply:
Oh just read your argument about why we would want it outside of plutus-apps. However, with CHaP, is it still relevant? I believe that if plutus-apps only uses released version of ledger and node, we can use different versions depending on the components in plutus-apps.

Reply:
Alright, if there are no limitations we go with plutus-apps.